Added homework lesson21.

iBubbleGun opened the pull request:
Homework lesson21 added.
Please, check it.

iBubbleGun committed:
Added homework lesson21.

iBubbleGun committed:
Updated homework lesson21.
Improved code readability.

kartius commented on src/main/java/edu/hillel/homework/lesson21/dao/DataBaseConnection.java:
```diff
@@ -0,0 +1,34 @@
+        return DriverManager.getConnection(db_url, db_username, db_password);
+    }
+
+    public static void close(Connection connection) throws SQLException {
```
Please, implement AutoClosable interface by this class and override close method to close connection properly

iBubbleGun replied:
Well done.

kartius commented on src/main/java/edu/hillel/homework/lesson21/dao/HomeworkDao.java:
```diff
@@ -0,0 +1,90 @@
+public class HomeworkDao {
+
+    public void addHomework(@NotNull Homework homework) {
+        try (Connection connection = DataBaseConnection.getConnection();
```
Please, send connection via constructor, the HomeworkDao shouldn't know about DataBaseConnection, it should be incapsulated

iBubbleGun replied:
Well done.

kartius commented on src/main/java/edu/hillel/homework/lesson21/dao/LessonDao.java:
```diff
@@ -0,0 +1,94 @@
+public class LessonDao {
+
+    public void addLesson(@NotNull Lesson lesson) {
+        try (Connection connection = DataBaseConnection.getConnection();
```
The same here

iBubbleGun replied:
Well done.

iBubbleGun committed:
Updated homework lesson21.
Improved implementation of database connection.
Added implementation of the AutoCloseable interface.

iBubbleGun committed:
Slight improvement to the use of the DataBaseConnection class.
